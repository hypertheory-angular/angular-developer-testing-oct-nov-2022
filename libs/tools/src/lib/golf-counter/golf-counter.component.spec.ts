import { Spectator, createComponentFactory } from '@ngneat/spectator/jest';
import { GolfCounterComponent } from './golf-counter.component';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import * as fromGolfCounterReducer from '../state/reducers/counter.reducer';
import * as rootState from '../state';
describe('GolfCounterComponent', () => {
  let spectator: Spectator<GolfCounterComponent>;
  let store: MockStore;
  const initialState = {
    counter: {
      count: fromGolfCounterReducer.reducer,
    },
  };

  const createComponent = createComponentFactory({
    component: GolfCounterComponent,
    providers: [
      provideMockStore({
        initialState,
        selectors: [
          {
            selector: rootState.selectCurrentCount,
            value: 444,
          },
        ],
      }),
    ],
  });

  beforeEach(() => {
    spectator = createComponent();
    store = spectator.inject(MockStore);
    spectator.detectChanges();
  });

  it('Initial State', () => {
    // spectator.detectChanges();
    // const component = spectator.query<HTMLSpanElement>(
    //   '[data-testid="current"]',
    // );
    // expect(component).toBeDefined();

    // console.log(component);
    // expect(component?.innerText).toBe('99');
    const component = spectator.query('[data-testid="counter-component"]');
    expect(component).toMatchSnapshot();
  });
});
