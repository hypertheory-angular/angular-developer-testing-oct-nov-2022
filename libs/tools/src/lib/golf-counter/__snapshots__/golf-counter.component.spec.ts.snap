// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GolfCounterComponent Initial State 1`] = `
<div
  data-testid="counter-component"
>
  <button
    class="btn btn-primary"
    data-testid="decrement-button"
  >
     Decrease
  </button>
  <span
    data-testid="current"
  >
    444
  </span>
  <button
    class="btn btn-primary"
    data-testid="increment-button"
  >
     + 
  </button>
</div>
`;
